src/components/form/form.html

```diff
@@ -1,19 +1,21 @@
-<form :id="form.id"
+<form class="m-form"
+      :id="form.id"
       @submit.prevent="submit"
       @reset.prevent="reset">
-    <m-message class="m-u--margin-bottom--l"
+    <m-message class="m-form__message-summary"
                v-if="hasErrors"
                :state="messageStateEror"
                ref="summary">
-        <ul class="m-u--no-margin">
-            <li v-for="error in errors">{{ error }}</li>
+        <ul>
+            <li v-for="error in errors"
+                v-html="error"></li>
         </ul>
     </m-message>
 
-    <p class="m-u--require"
-       v-if="hasRequiredFields">
-        <span class="m-u--asterisk">*</span>
-        <span class="m-u--typo--precision">{{ 'm-form:required' | f-m-i18n }}</span>
+    <p class="m-form__required"
```

Comment:
Le block ici c'est plutôt le 'form-required' ou même juste 'required', avec 2 éléments 'marker' et 'label'.
Donc:
form-required
form-required__marker
form-required__label
C'est basé sur http://getbem.com/naming/

Reply:
mais l'element est bien un element de m-form, donc la syntax ne serait pas m-form__required__marker? (required est un element de m-form, marker est un element de required) @raphpare  c'est quoi ton avis la dessus?

Reply:
mais l'element required est bien un element de m-form, donc la syntax ne serait pas m-form__required__marker? (required est un element de m-form, marker est un element de required) @raphpare c'est quoi ton avis la dessus?

Reply:
En fait faut pas forcément que tu fasse ton BEM en fonction du markup,  C'est pas parce qu'un élément A est contenu dans l'élément B que dans BEM tu auras B__A

Le concept de requis est un concept qui peut être standalone, qui fait du sens en lui même.  Alors, comme Alvéric dit, on peut avoir un block required qui contient un label et un marker.

Reply:
deal! j'update ça!

Reply:
Il faut bien comprendre que nous ne faisons pas du BEM.

Il faut donc absolument conserver la nomenclature `m-form__required`, si non on ne sais pas d'où vient la classe CSS lorsqu'on débogue. 

Il faut jamais faire de classe standalone dans les composants. Les classes doivent toujours commencer par le nom du composant `m-non-composant__element-composant` à l'execption des classe modifier `m--[is|has|no]-modification`.

Voir https://wiki.dti.ulaval.ca/pages/viewpage.action?pageId=70559033

Reply:
Pourtant on a eu une formation front-end sur le BEM.. si on en fait pas pourquoi en parler T_T

Reply:
je propose de suivre notre documentation. Si vous pensez pertinents de changer la doc, continuez le débat. Ça peut être pertinent. Mais pour le moment je vais remettre la classe dans m-form comme suggère @raphpare.

Reply:
Le BEM est utile lorsque les styles ne sont pas scopé et peuvent être réutilisé. Dans notre cas les classes qui peuvent être réutilisé se nomme "classes utilitaires" et leur nomenclature est la suivante `m-u--nom-classe`. Les classes utilitaires ne peuvent pas être utilisées dans les composants modul, car on veut que les composants fonctionnent sans les styles de base de modul (main.scss).